src/agent/agent.h

```diff
@@ -82,6 +83,7 @@ struct Agent {
         sd_bus *peer_dbus;
 
         bool metrics_enabled;
+        sd_bus_slot *register_call_slot;
         sd_bus_slot *metrics_slot;
```

Comment:
NIT: The `register_call_slot` isn't really related to the metrics, so I'd keep it separate.
```suggestion
        sd_bus_slot *register_call_slot;

        bool metrics_enabled;
        sd_bus_slot *metrics_slot;
```

Reply:
Right. Mess created by rebase